### ISSUE TITLE
Face Recognition

### ISSUE THREAD

**Issue by linahourieh** [linahourieh/machine-learning-articles#10]
## TL;DR
Embeddings , triplet loss

### Article Link
FaceNet: A Unified Embedding for Face Recognition and Clustering

### Author
Google Researchers 

## Key Takeaways
* based on learning a Euclidean embedding per image using a deep convolutional network.

![image](https://user-images.githubusercontent.com/81252980/189308456-bb7f8fe2-52eb-4ad5-ba11-dcd24a2365a0.png)

***face verification** simply involves thresholding the distance between the two embeddings;
***face recognition** becomes a k-NN classification problem;

### Triplet Selection
* We have to remember that our aim is to decrease d(A,P) & increase d(A,N) with a margin of alpha. On that basis, we update our parameters. Now, there is a high chance that the d(A,N) would be higher than d(A,P). Then, there wouldn't b much room for improvement regarding our parameters. Then, the update of the update of the parameters won't be significant and the convergence would be slow. 
* However, if we choose a pair where d(A,N) is low and d(A,P) is high, then our parameters would be updated with great margin that allows good convergence. 
![Untitled presentation (1)](https://user-images.githubusercontent.com/81252980/189205180-ed4405b8-1976-44e0-a9e8-2e4789e749cc.jpg)
![Untitled presentation](https://user-images.githubusercontent.com/81252980/189205183-c1de0de4-284b-4b58-a566-a5c0f34ad8cc.jpg)

https://arsfutura.com/magazine/face-recognition-with-facenet-and-mtcnn/
